src/routers/UpdateMatchAvailabilitiesRouter.ts

```diff
@@ -0,0 +1,80 @@
+import { Request } from 'express';
+import Constants from '../common/constants';
+import Availability from '../entities/Availability';
+import AvailabilityRepo from '../repos/AvailabilityRepo';
+import MatchRepo from '../repos/MatchRepo';
+import AuthenticatedApplicationRouter from '../utils/AuthenticatedApplicationRouter';
+
+class UpdateMatchAvailabilitiesRouter extends AuthenticatedApplicationRouter<void> {
+  constructor() {
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/availabilities/';
+  }
+
+  async content(req: Request): Promise<void> {
+    const { matchID, schedule } = req.body;
+
+    if (!matchID) throw Error('Missing required matchID');
+    if (!schedule) throw Error('Missing required schedule list');
+
+    // Logic to update match status
+    const match = await MatchRepo.getMatchByID(matchID);
+    let status = Constants.MATCH_CREATED;
+    if (match.status !== Constants.MATCH_CREATED && match.status !== Constants.MATCH_PROPOSED) {
+      throw Error('Cannot update match availability at this stage!');
+    }
+    if (match.status === Constants.MATCH_CREATED) {
+      status = Constants.MATCH_PROPOSED;
+    }
+    if (match.status === Constants.MATCH_PROPOSED) {
+      status = Constants.MATCH_ACTIVE;
+      if (schedule.length !== 1 || schedule[0].times.length !== 1) {
+        throw Error('Schedule must contain exactly one day and one time to finalize the match!');
```

Comment:
just wanted to note this is probably the best way to do this because I thought why not enforce this in what frontend needs to pass back, but then they'd need to make a completely new model for this single route